### ISSUE TITLE
followRelationships to follow nested relations

### ISSUE THREAD

**Issue by staaam** [mrichar1/jsonapi-vuex#22]
Currently followRelationships expands included object only to one level.
According to JSON:API specs, it's possible to include multiple levels of relations like this: /articles/1?include=comments.author (https://jsonapi.org/format/#fetching-includes).

I'd like to propose to make followRelationships recursive to support this. I already have some draft for this so can create PR with the code.

**Comment by mrichar1** [comment]
Thanks for this. Yes, recursion would be nice (if done safely!) Happy to consider any suggestions.

The codebase is in a bit of flux just now (half of which is in the `status` branch) so best hold off any PRs until that's been merged back into master and a new release made (hopefully some time next week).

**Comment by mhluska** [comment]
This is a major one for me as I'm starting to use nested resources. I could help with this one unless you've got something ready @staaam ?

**Comment by mrichar1** [comment]
Great! I've literally just pushed some major refactoring (still to the `status` branch) (TL:DR; `action` strings are treated like URLs, `getRelated` now has properly chained, not nested (anti-pattern!) promises, test suite tidying/fixing).

@mhluska if you could check that the original 'status id' work hasn't been broken by all of this, I'll push this into master, and then it should be a good point to start looking at relationship recursion. (and I'll revisit the other outstanding issues I said I was happy to develop).

**Comment by mhluska** [comment]
@mrichar1 the status seems to be working fine for me on the latest commit. But admittedly I'm not relying on it heavily yet. Still building out many areas of the front end.

**Comment by staaam** [comment]
Yeah, I already have pretty working version, based on current master though. I'll try to push it today to my github, and start looking at how it's affected by status branch

**Comment by mrichar1** [comment]
That's `status` merged into master and released as `0.1.0` (nothing/very little has changed in `followRelationships`) so happy to see a PR/work towards this enhancement.

**Comment by mhluska** [comment]
@staaam just curious where you're at with the nested relationships stuff? Looks like you have a fork with some progress?

**Comment by staaam** [comment]
Sorry, just finally got to finishing it. I've opened PR https://github.com/mrichar1/jsonapi-vuex/pull/28 . Main change is it followRelationships which instead of cloning related objects on inner loop makes recursive call to itself. I've added follow_state to prevent infinite recursion which holds all objects which was visited during current followRelationships traversing.
One test was fixed (should return normalized data with expanded rels (array)) - the expected norm_record_rels is now recursive.

Any comments/suggestions?
